### ISSUE TITLE
Misleading error when using ufuncs on large python ints

### ISSUE THREAD

**Issue by MaxNoe** [numpy/numpy#8357]
```
In [2]: import numpy as np
In [2]: np.log10(2**10)
Out[2]: 3.0102999566398121
In [3]: np.log10(2**64)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-3-a2d6ba68c7f4> in <module>()
----> 1 np.log10(2**64)

AttributeError: 'int' object has no attribute 'log10
```

I guess this is because python's unlimited precision int cannot be transformed into a 64 bit unsigned int anymore. But the error message is very misleading on this.

**Comment by charris** [comment]
Really, the only other thing to do is raise an error message when the number is converted to an object array. The automatic conversion of some types to object arrays causes other problems as well, and we may change that behavior at some point.

**Comment by MaxNoe** [comment]
I fully agree that this should raise an error instead of silently converting to a float number. But the error message should be a more specific.

**Comment by charris** [comment]
What would you suggest? The number gets converted to an object array that contains a long integer object. Since all types of objects can be in object arrays, the log function doesn't know how to handle it besides looking for an appropriate object method to call, which is missing in this case, and that is what the error message tells you. It is only confusing if you don't know what is going on, but the function has no idea how you got to the point where it is called, so cannot help you much in that regard.

**Comment by eric-wieser** [comment]
I think that in general, calling `np.func` on a object array should wrap the error message. So for instance, something like:

```python
>>> class Foo: pass
>>> np.log10(Foo())
TypeError: Cannot call log10 on an object array when the element <__main__.Foo object at 0x000001EC87E80B00> has no 'log10' attribute
```

**Comment by eric-wieser** [comment]
Fixed in #12700
